src/controller/CHIPDeviceControllerFactory.cpp

```diff
@@ -147,31 +147,36 @@ CHIP_ERROR DeviceControllerFactory::InitSystemState(FactoryInitParams params)
 #endif
                                                             ));
 
-    stateParams.fabricTable                                   = chip::Platform::New<FabricTable>();
     stateParams.sessionMgr                                    = chip::Platform::New<SessionManager>();
     SimpleSessionResumptionStorage * sessionResumptionStorage = chip::Platform::New<SimpleSessionResumptionStorage>();
     stateParams.sessionResumptionStorage                      = sessionResumptionStorage;
     stateParams.exchangeMgr                                   = chip::Platform::New<Messaging::ExchangeManager>();
     stateParams.messageCounterManager                         = chip::Platform::New<secure_channel::MessageCounterManager>();
     stateParams.groupDataProvider                             = params.groupDataProvider;
 
-    ReturnErrorOnFailure(stateParams.fabricTable->Init(params.fabricIndependentStorage));
+    // if no fabricTable was provided, create one and track it in stateParams for cleanup
+    FabricTable * fabricTable = params.fabricTable;
+    if (fabricTable == nullptr)
+    {
+        stateParams.fabricTable = fabricTable = chip::Platform::New<FabricTable>();
```

Comment:
OK, but if `fabricTable` is not null, then `stateParam.fabricTable` ends up null, and then the `DeviceControllerSystemState` will return null from `Fabrics`, which will probably cause crashes somewhere around `DeviceController::ProcessControllerNOCChain`.

The thing we probably want to do is:
1. Set the fabric table on the system state.
2. Tell the system state to not delete the table.
3. Store the table in a member of ours, so if we go through the no-args InitSystemState later (e.g. if we shut down the last controller and then try to start a new one) we can set it up on the system state properly...